### ISSUE TITLE
Ingot former doesn't produce correct ingot count when storage only has 1 ingot available to add

### ISSUE THREAD

**Issue by Boomer-1** [Mooy1/InfinityExpansion#103]
## Description (Required)

When storing ingots in any storage unit, and 1 ingot is pulled out, the ingot former incorrectly thinks there is enough room to run another cycle of 8 dust to 4 ingots. As it creates the 4 ingots, only 1 actually is created and sent to the storage unit instead of 4. Because there isn't enough room in the storage for the other 4, 3 are deleted instead of remaining in the ingot former.

## Expected behavior (Required)

the ingot former to either wait until there is enough room to add 4 ingots, or keep the remaining 3 in the machine until there is room.

## Server Log / Error Report

none

## Environment (Required)

 - Minecraft Version: 1.18.1
 - Slimefun Version: 1006
 - Plugin Version: 137

**Comment by Sefiraat** [comment]
@Boomer-1 Is this still an issue and would you still consider it a bug as opposed to a disliked-feature?

**Comment by Boomer-1** [comment]
yes it is a bug still. A few others have reported it as well in discord chat.

**Comment by Sefiraat** [comment]
So, this is alhappenkng as a by product of IE using infinitylib’s ID/name getter which is linked to to many things at the moment to be replaced without causing more harm than good. I think this only effects copper and composters (composter singularity confirmed). I don’t want to add a manual fix to the machine itself for this one case as I feel it’s relatively harmless. I have made a note in the flaw with the specific method and will fix when I have more of a grounding in the code base. For now this issue will be open for some time.

**Comment by JasperChaseTOQ** [comment]
i think wait until there's room for 4 would be best,  because otherwise it'll flood cargo and spit the items out
